src/java/org/apache/cassandra/repair/autorepair/AutoRepairState.java

```diff
@@ -0,0 +1,349 @@
+    public int getNodeRepairTimeInSec()
+    {
+        return nodeRepairTimeInSec;
+    }
+
+    public void setRepairInProgress(boolean repairInProgress)
+    {
+        this.repairInProgress = repairInProgress;
+    }
+
+    public boolean isRepairInProgress()
+    {
+        return repairInProgress;
+    }
+
+    public void setRepairSkippedTablesCount(int count)
+    {
+        repairTableSkipCount = count;
+    }
+
+    public int getRepairSkippedTablesCount()
+    {
+        return repairTableSkipCount;
+    }
+
+    public int getLongestUnrepairedSec()
+    {
+        if (longestUnrepairedNode == null)
+        {
+            return 0;
+        }
+        return (int) TimeUnit.MILLISECONDS.toSeconds(timeFunc.get() - longestUnrepairedNode.getLastRepairFinishTime());
+    }
+
+    public void setRepairFailedTablesCount(int count)
+    {
+        repairTableFailureCount = count;
+    }
+
+    public int getRepairFailedTablesCount()
+    {
+        return repairTableFailureCount;
+    }
+
+    public void setTotalMVTablesConsideredForRepair(int count)
+    {
+        totalMVTablesConsideredForRepair = count;
+    }
+
+    public int getTotalMVTablesConsideredForRepair()
+    {
+        return totalMVTablesConsideredForRepair;
+    }
+
+    public void setNodeRepairTimeInSec(int elapsed)
+    {
+        nodeRepairTimeInSec = elapsed;
+    }
+
+    public void setClusterRepairTimeInSec(int seconds)
+    {
+        clusterRepairTimeInSec = seconds;
+    }
+
+    public void setRepairKeyspaceCount(int count)
+    {
+        repairKeyspaceCount = count;
+    }
+
+    public void setRepairTableSuccessCount(int count)
+    {
+        repairTableSuccessCount = count;
+    }
+
+    public int getRepairKeyspaceCount()
+    {
+        return repairKeyspaceCount;
+    }
+
+    public int getRepairTableSuccessCount()
+    {
+        return repairTableSuccessCount;
+    }
+
+    public void setLongestUnrepairedNode(AutoRepairHistory longestUnrepairedNode)
+    {
+        this.longestUnrepairedNode = longestUnrepairedNode;
+    }
+
+    public boolean isSuccess()
+    {
+        return success;
+    }
+
+    public void recordTurn(AutoRepairUtils.RepairTurn turn)
+    {
+        metrics.recordTurn(turn);
+    }
+
+    public void setTotalDisabledTablesRepairCount(int count)
+    {
+        totalDisabledTablesRepairCount = count;
+    }
+
+    public int getTotalDisabledTablesRepairCount()
+    {
+        return totalDisabledTablesRepairCount;
+    }
+
+    public void resetWaitCondition()
+    {
+        condition = newOneTimeCondition();
+    }
+}
+
+class IncrementalRepairState extends AutoRepairState
+{
+    public IncrementalRepairState()
+    {
+        super(RepairType.incremental);
+    }
+
+    @Override
+    public RepairRunnable getRepairRunnable(String keyspace, List<String> tables, Set<Range<Token>> ranges, boolean primaryRangeOnly)
+    {
+        RepairOption option = new RepairOption(RepairParallelism.PARALLEL, primaryRangeOnly, true, false,
```

Comment:
Discussion point:  We are currently passing `false` for `optimiseStreams`.  This is a nice enhancement in the form of [`CASSANDRA-3200`](https://issues.apache.org/jira/browse/CASSANDRA-3200) which cuts down the amount of overstreaming when inconsistencies are found.

I'm not sure it has a lot of precedence for use, but it's been in tree for a long time.  We've only started to look into rolling it out ourselves.

I think we should consider setting it to true here, or making it configurable somehow.

Reply:
Ok, changed it to `true`